tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/DebugTestRunner.java

```diff
@@ -588,14 +587,36 @@ public void assertExpression(StoppedEventArguments context, String expression, S
     public void assertEvaluationError(StoppedEventArguments context, String expression, String errorMessage)
             throws BallerinaTestException {
         Variable result = evaluateExpression(context, expression);
-        switch (assertionMode) {
-            case HARD_ASSERT:
+        if (assertionMode == AssertionMode.HARD_ASSERT) {
+            if (result.getType() == null) {
+                Pair<String, OutputEventArguments> output = waitForDebugOutput(2000);
+                String errorOutput = output.getLeft();
+                if (errorOutput.endsWith(System.lineSeparator())) {
+                    errorOutput = errorOutput.replaceAll(System.lineSeparator() + "$", "");
+                }
+                Assert.assertNull(result.getValue());
+                Assert.assertNull(result.getType());
+                Assert.assertEquals(output.getRight().getCategory(), OutputEventArgumentsCategory.STDERR);
+                Assert.assertEquals(errorOutput, errorMessage);
+            } else {
+                Assert.assertEquals(result.getType(), "error");
                 Assert.assertEquals(result.getValue(), errorMessage);
-                Assert.assertTrue(result.getType().equals("string") || result.getType().equals("error"));
-                return;
-            case SOFT_ASSERT:
+            }
```

Comment:
Repeated code block. Shall we add this to a utility function and pass the  `Assert` object as a parameter?